### PR TITLE
SD-573 Add Post Put  and Delete functionality

### PR DESCRIPTION
**Pull request by buzz1710**

### CODE REVIEW

File: src/org/sage_one_sample/sageone/SageoneData.java [hunk 1]
```diff
@@ -1,77 +1,175 @@
...
+		String signature = s.signature();
+
+		//build the request
+		HttpClient httpclient = HttpClients.createDefault();
+		try {
+			URIBuilder builder = new URIBuilder(endpoint);
+			URI uri = builder.build();
+			HttpRequestBase request = requestMethod.equals("GET") ? new HttpGet(uri) : new HttpDelete(uri);
+
+			// set the request headers
```
Comment (adamgeorgeson) [inline src/org/sage_one_sample/sageone/SageoneData.java]: Can this be moved into it's own method?

File: src/org/sage_one_sample/sageone/SageoneData.java [hunk 2]
```diff
@@ -1,77 +1,175 @@
...
+		{
+			System.out.println(e.getMessage());
+		}
+	}
+
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Comment (adamgeorgeson) [inline src/org/sage_one_sample/sageone/SageoneData.java:74]: maybe clearer that this is doing both PUT and POST?

File: src/org/sage_one_sample/sageone/SageoneToken.java [hunk 3]
```diff
@@ -6,77 +6,142 @@
...
+			resp.getWriter().println("</html>");
+		} catch (IOException ex) {
+			ex.printStackTrace();
+		}
+	}
+
+	/**
+	 * @see HttpServlet#doPost(HttpServletRequest request, HttpServletResponse response)
+	 */
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (adamgeorgeson) [inline src/org/sage_one_sample/sageone/SageoneToken.java:146]: Not sure I get this.

Comment (buzz1710) [inline src/org/sage_one_sample/sageone/SageoneToken.java:146]: this is how a servlet is generated by default - 2 methods `doGet` and `doPost` with `doPost` simply calling `doGet`. Not entirely sure why though.

**Comment by adamgeorgeson** [comment]
👍

File: src/org/sage_one_sample/sageone/Nonce.java [hunk 4]
```diff
@@ -7,21 +7,21 @@
...
+			if (secureRandom.nextBoolean()) {
+				offset = 48;
+				range = 10;
+			} else {
+				offset = 97;
+				range = 6;
+			}
+			bytes[i] = (byte) (offset + secureRandom.nextInt(range));
+		}
+		return new String(bytes);
```
Comment (SeanArmstrong) [inline src/org/sage_one_sample/sageone/Nonce.java:25]: I thought you had to do `Array.toString(bytes)` but if `String(bytes)` works thats ok

File: src/org/sage_one_sample/sageone/SageoneData.java [hunk 5]
```diff
@@ -1,77 +1,167 @@
...
-  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-    doGet(request, response);
-  }
+	private static final long serialVersionUID = 1L;
+
+	/* GET and DELETE requests */
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+		String requestMethod = req.getParameter("request_method").toUpperCase();
+		String endpoint = "https://api.sageone.com/" + req.getParameter("endpoint");
+		TreeMap<String, String> params = new TreeMap<String,String>();
```
Comment (SeanArmstrong) [inline src/org/sage_one_sample/sageone/SageoneData.java:44]: missing space between `String,String`

**Comment by SeanArmstrong** [comment]
Not stopping the PR but would a sample code example require more comments to address what certain lines or functions are doing. Hopefully making the experience easier for developers

**Comment by SeanArmstrong** [comment]
👍

**Pull request merged**